src/definitions/x509/extensions.rs

```diff
@@ -0,0 +1,380 @@
+                    errors.push(Error::ValidationError(format!("crl_issuer and reasons may not be set on CrlDistributionPoints, but is set for: {:?}", point)))
+                }
+
+                if !point
+                    .distribution_point
+                    .clone()
+                    .is_some_and(|dpn| match dpn {
+                        DistributionPointName::FullName(names) => {
+                            let type_errors: Vec<Error> = check_general_name_types(names);
+                            type_errors.is_empty()
+                        }
+                        DistributionPointName::NameRelativeToCRLIssuer(_) => false,
+                    })
+                {
+                    errors.push(Error::ValidationError(format!(
+                        "crl distribution point has an invalid type: {:?}",
+                        point
+                    )))
+                }
+            }
+        }
+        Err(e) => errors.push(Error::DecodingError(e.to_string())),
+    }
+
+    errors
+}
+
+/*  The Issuer Alternative Name must be of a type URI or RFC822Name */
+pub fn validate_issuer_alternative_name(bytes: Vec<u8>) -> Vec<Error> {
+    let iss_altname = IssuerAltName::from_der(&bytes);
+    match iss_altname {
+        Ok(ian) => check_general_name_types(ian.0),
+        Err(e) => {
+            vec![Error::DecodingError(e.to_string())]
+        }
+    }
+}
+
+/*  Basic Constraints must be CA: true, path_len: 0 */
+pub fn validate_basic_constraints(bytes: Vec<u8>) -> Vec<Error> {
+    let basic_constraints = BasicConstraints::from_der(&bytes);
+    match basic_constraints {
+        Ok(bc) => {
+            if !bc.path_len_constraint.is_some_and(|path_len| path_len == 0) && bc.ca {
+                return vec![Error::ValidationError(format!(
+                    "Basic constraints expected to be CA:true, path_len:0, but found: {:?}",
+                    bc
+                ))];
+            }
+            vec![]
+        }
+        Err(e) => {
+            vec![Error::DecodingError(e.to_string())]
+        }
+    }
+}
+
+fn check_general_name_types(names: Vec<GeneralName>) -> Vec<Error> {
+    let valid_types: Vec<bool> = names
+        .iter()
+        .map(|name| {
+            matches!(
+                name,
+                GeneralName::Rfc822Name(_) | GeneralName::UniformResourceIdentifier(_)
+            )
+        })
+        .collect();
+
+    if valid_types.contains(&false) {
+        vec![Error::ValidationError(format!(
+            "Invalid type found in GeneralNames: {:?}",
+            names
+        ))]
+    } else {
+        vec![]
+    }
+}
+
+pub fn iaca_disallowed_x509_extensions() -> Vec<String> {
+    vec![
+        "2.5.29.30".to_string(),
+        "2.5.29.33".to_string(),
+        "2.5.29.36".to_string(),
+        "2.5.29.46".to_string(),
+        "2.5.29.54".to_string(),
+    ]
+}
+
+#[cfg(test)]
+pub mod test {
+
+    #[test]
+    fn test_key_usage() {}
```

Comment:
Could you write some unit tests or just remove this. Ideally adding some unit tests would be great if you have time.